contracts/Remittance.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity 0.5.10;
+
+import "./SafeMath.sol";
+import "./Stoppable.sol";
+
+contract Remittance is Stoppable {
+    using SafeMath for uint;
+    
+    struct Order {
+        uint deadline;
+        address emitter;
+        uint amount;
+    }
+    
+    uint constant TX_FEES = 2000;
+
+    mapping(bytes32 => Order) public orders;
+    mapping(address => uint) public fees;
+    
+    event LogNewOrder(address indexed emitter, uint amount, bytes32 hashedOTP);
+    event LogTakeFees(address indexed emitter, uint amount);
+    event LogCancelOrder(address indexed emitter, uint amount, bytes32 hashedOTP);
+    event LogWithdraw(address indexed emitter, uint amount);
```

Comment:
`hashedOTP`?

Reply:
Also, do you think people will filter per hashedOTP from the client side?

Reply:
To track undergoing transfers yes, I think.